apps/backend/src/lib/rds.ts

```diff
@@ -22,6 +31,106 @@ export class RDS {
             .then((xs) => xs[0]?.id ?? null);
     }
 
+    /**
+     * Retrieves an account with the specified user ID and account type 'GOOGLE'.
+     *
+     * @param db - The database or transaction object.
+     * @param userId - The ID of the user whose account is to be retrieved.
+     * @returns A promise that resolves to the account object if found, otherwise null.
+     */
+    static async getAccountByProviderId(
+        db: DatabaseOrTransaction,
+        accountType: 'GOOGLE' | 'GUEST',
```

Comment:
This should be imported from `account`. 

Also, the docstring doesn't match what this function does, and this is a generalization of `guestUserIdWithNameOrNull`, so that function should be refactored to call this one.